### PR TITLE
[Messenger][DoctrineBridge] Extract Abstract Doctrine Middleware

### PR DESCRIPTION
**Pull request by Koc**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no 
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Just a minior refactoring to prevent code duplicate.

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/Messenger/AbstractDoctrineMiddleware.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Exception\UnrecoverableMessageHandlingException;
+use Symfony\Component\Messenger\Middleware\MiddlewareInterface;
+use Symfony\Component\Messenger\Middleware\StackInterface;
+
+/**
+ * @author Konstantin Myakshin <molodchick@gmail.com>
+ *
+ * @experimental in 4.4
```
Comment (chalasr) [inline src/Symfony/Bridge/Doctrine/Messenger/AbstractDoctrineMiddleware.php]: The component must be stable for 4.4 (LTS) :)

Comment (derrabus) [inline src/Symfony/Bridge/Doctrine/Messenger/AbstractDoctrineMiddleware.php]: Maybe we should use `@internal` instead.

Comment (Koc) [reply src/Symfony/Bridge/Doctrine/Messenger/AbstractDoctrineMiddleware.php]: done

**Comment by Koc** [comment]
I've switched branch to 4.4, tests are green.

**Review by derrabus — APPROVED** [review]

**Comment by Koc** [comment]
PR now includes `DoctrineClearEntityManagerMiddleware` as well.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Koc.

**Pull request merged**